modules/apps/frontend-js/frontend-js-a11y-web/src/main/resources/META-INF/resources/A11yPanel.tsx

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import React, {useState} from 'react';
+
+import Occurrence from './Occurrence';
+import Violation from './Violation';
+import Violations from './Violations';
+import {FilteredViolationsContextProvider} from './useFilteredViolations';
+
+import type {ImpactValue, Result} from 'axe-core';
+
+function isBetweenRange(childrenCount: number, newIndex: number) {
+	return 0 <= newIndex && newIndex <= childrenCount;
+}
+
+type SidebarPanelNavigatorProps = {
+	children: React.ReactNode;
+};
+
+type TNavigationState = {
+	occurrenceIndex?: number;
+	occurrenceName?: string;
+	violationsIndex: number;
+};
+
+enum PagesEnum {
+	Occurrence = 2,
+	Violation = 1,
+	Violations = 0,
+}
+
+function SidebarPanelsNavigator({children}: SidebarPanelNavigatorProps) {
+	const [activePageIndex, setActivePageIndex] = useState(
+		PagesEnum.Violations
+	);
+
+	const [navigationState, setNavigationState] = useState<TNavigationState>({
+		violationsIndex: 0,
+	});
```

Comment:
Hey @diegonvs in these cases we should try to abstract the data from the Navigator, the idea is that it does not know what data it supports or is rendering.

For example, it doesn't make much sense to pass `PagesEnum.Violations` as the initial property,`enum` will only be used here only once, so the use of `enum` is not necessary, so just setting it to `0` will be enough.

Having an initial value for the state is also a little strange because we will only have initial values when you browse, they are navigation states, in this case, create a generic type instead of assigning `TNavigationState` to the state. You don't need that value for the component, right?

Reply:
Makes sense removing PagesEnum here because it carry the old logic of deciding where to navigate based on keys, different from the `next` and `previous` approach here. I'll create a constant called VIOLATIONS_PAGE_INDEX and attribute the index 0 to it. Is it okay?

Reply:
I would avoid creating a constant for this, it is not necessary, because `0` could be anything in this case, so within the context of the Navigator, `0` will be the first element.

Reply:
Yes, depends of the order of the elements passed as a children of `SidebarPanelsNavigator` component. Removing the constant!